src/store/modules/chats.js

```diff
@@ -677,27 +677,49 @@ export default {
           const outpointsList = stealthMessage.getOutpointsList()
           const ephemeralPubKeyRaw = stealthMessage.getEphemeralPubKey()
           const ephemeralPubKey = PublicKey.fromBuffer(ephemeralPubKeyRaw)
+          const stealthHDPrivKey = constructStealthPrivKey(ephemeralPubKey, identityPrivKey)
           let totalSatoshis = 0
-          for (const [, outpoint] of Object.entries(outpointsList)) {
+          for (const [i, outpoint] of Object.entries(outpointsList)) {
             const stealthTxRaw = Buffer.from(outpoint.getStealthTx())
             const stealthTx = cashlib.Transaction(stealthTxRaw)
             const txId = stealthTx.hash
             const vouts = outpoint.getVoutsList()
-            for (const outputIndex of vouts) {
+            for (const [j, outputIndex] of Object.entries(vouts)) {
               const output = stealthTx.outputs[outputIndex]
               const satoshis = output.satoshis
-              totalSatoshis += satoshis
+
               if (!outbound) {
-                continue
                 // We don't want to add these to the wallet, but we do want the total
+                // We also can't compute the private key.... So the below address check would
+                // error
+
+                // Assume our output was correct and add to the total
+                totalSatoshis += satoshis
+                continue
               }
-              const outputPrivKey = constructStealthPrivKey(ephemeralPubKey, identityPrivKey)
-              const address = output.script.toAddress('testnet').toLegacyAddress() // TODO: Make generic
+              const outpointPrivKey = stealthHDPrivKey
+                .deriveChild(44)
+                .deriveChild(145)
+                .deriveChild(i)
+                .deriveChild(j)
+                .privateKey
+              const address = output.script.toAddress('testnet') // TODO: Make generic
```

Comment:
We can probably take a shortcut here and do `output.script.getPublicKeyHash()`?

https://github.com/bitpay/bitcore/blob/5dfeb941eaca5e22afa265725d9304cdf9e2fd57/packages/bitcore-lib/lib/script/script.js#L351

Reply:
There's no way to get that for the PublicKey.